internal/clients/http/gcp/gcp_client.go

```diff
@@ -164,41 +162,90 @@ func (c *gcpClient) InsertInstances(ctx context.Context, params *clients.GCPInst
+	}
+	ids, err := c.ListInstancesIDsByTag(ctx, params.UUID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("cannot list instances ids: %w", err)
+	}
+
+	return ids, ptr.To(op.Name()), nil
+}
+
+func (c *gcpClient) ListInstancesIDsByTag(ctx context.Context, uuid string) ([]*string, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "listInstancesIDs")
+	defer span.End()
+
+	logger := logger(ctx)
+	ids := make([]*string, 0)
 	filter := fmt.Sprintf("labels.rhhcc-id=%v", uuid)
 	lstReq := &computepb.AggregatedListInstancesRequest{
 		Project: c.auth.Payload,
 		Filter:  &filter,
 	}
 
-	if !op.Done() {
-		return nil, ptr.To(op.Name()), fmt.Errorf("an error occured on operation %s: %w", op.Name(), ErrOperationFailed)
+	client, err := c.newInstancesClient(ctx)
+	if err != nil {
+		logger.Error().Err(err).Msg("Could not get instances client")
+		return nil, fmt.Errorf("unable to get instances client: %w", err)
 	}
+	defer client.Close()
 
-	ids := make([]*string, 0)
 	instances := client.AggregatedList(ctx, lstReq)
+	logger.Trace().Msg("Fetching instance ids")
 	for {
 		pair, err := instances.Next()
 		if errors.Is(err, iterator.Done) {
-			logger.Error().Err(err).Msg("Instances iterator has finished")
+			logger.Trace().Err(err).Msg("Instances iterator has finished")
 			break
 		} else if err != nil {
 			logger.Error().Err(err).Msg("An error occurred during fetching instance ids")
 			span.SetStatus(codes.Error, err.Error())
-			return nil, nil, fmt.Errorf("cannot fetch instance ids: %w", err)
+			return nil, fmt.Errorf("cannot fetch instance ids: %w", err)
 		} else {
-			logger.Trace().Msg("Fetching instance ids")
 			instances := pair.Value.Instances
-			for _, o := range instances {
-				idAsString := strconv.FormatUint(o.GetId(), 10)
+			for _, insta := range instances {
+				idAsString := strconv.FormatUint(insta.GetId(), 10)
 				ids = append(ids, &idAsString)
 			}
 		}
 	}
-	return ids, ptr.To(op.Name()), nil
+	return ids, nil
+}
+
+func (c *gcpClient) GetInstanceDescriptionByID(ctx context.Context, id string) (*clients.InstanceDescription, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "GetInstanceByID")
+	defer span.End()
+
+	logger := logger(ctx)
+
+	client, err := c.newInstancesClient(ctx)
+	if err != nil {
+		logger.Error().Err(err).Msg("Could not get instances client")
+		return nil, fmt.Errorf("unable to get instances client: %w", err)
+	}
+	defer client.Close()
+
+	projectId := c.auth.String()
+
+	instance, err := client.Get(ctx, &computepb.GetInstanceRequest{Instance: id, Project: projectId})
+	if err != nil {
+		return nil, fmt.Errorf("unable to get instance: %w", err)
+	}
+	instanceId := strconv.FormatUint(instance.GetId(), 10)
+	instanceDesc := clients.InstanceDescription{ID: instanceId}
+	for _, n := range instance.NetworkInterfaces {
+		if len(n.AccessConfigs) > 0 {
+			instanceDesc.PublicIPv4 = *n.AccessConfigs[0].NatIP
```

Comment:
I still have the initial concern, you dereference something that might be `nil`. This needs to be:

```go
if n.AccessConfigs[0] != nil {
  instanceDesc.PublicIPv4 = *n.AccessConfigs[0].NatIP
}
```